src/aof.c

```diff
@@ -600,8 +600,37 @@ sds catAppendOnlyGenericCommand(sds dst, int argc, robj **argv) {
     return dst;
 }
 
+/* Generate a piece of timestamp annotation for AOF if current record timestamp
+ * in AOF is not equal server unix time. If we specify 'force' argument to 1,
+ * we would generate one without check, currently, it is useful in AOF rewriting
+ * child process which always needs to record one timestamp at the beginning of
+ * rewriting AOF.
+ *
+ * Timestamp annotation format is "#TS:${timestamp}\r\n". "TS" is short of
+ * timestamp and this method could save extra bytes in AOF. */
+sds genAofTimestampAnnotationIfNeeded(int force) {
```

Comment:
Will the annotations generally be of the format: #{name}:{value}\r\n ?

Reply:
Also forgive my ignorance about AOF, but I thought it generally followed the RESP protocol, so instead should we use the attribute type notation from RESP3 which describes a similar syntax.

Reply:
Hi @madolson the format: #{name}:{value}\r\n is only one of annotations, the format is loose #9325.
Initially, i want to use a special command to identify annotations, but AOF is the set of write commands, so it will be odd, maybe we even add one more command for annotations, so i abandon.
The attribute type of RESP3 can represent notation, but i think it is for reply, i am not sure.

Reply:
I was reading this comment:
```
AOF (Append Only File): The AOF persistence logs every write operation received by the server, that will be played again at server startup, reconstructing the original dataset. Commands are logged using the same format as the Redis protocol itself, in an append-only fashion. Redis is able to rewrite the log in the background when it gets too big.
```

So if someone was tailing it, they could use a RESP parser to understand what was going on. Maybe that is a dumb comment and someone that runs a lot of AOF systems can tell me otherwise, I just wanted to mention it.

Reply:
i don't think we must look too much into that comment.
i think it is ok for the annotations to be AOF "comments" and not use RESP3 attributes.
First because AOF is indeed requests (which aren't actually using RESP3, just multi-bulks), and secondly, even if requests in the future will be using attributes, maybe we don't want to collide with them.
i.e. the annotations are (for now) generic comments about the AOF, and not explicitly referring to the immediate request that follows (next command).

so this actually raises another problem. what if AOF file will in the future wanna use the RESP3 booleans? (which use `#`)
maybe, just to be on the safe side, we should chose another character? or use two `##` instead of one?

Reply:
I think another character sounds reasonable.

Reply:
I think `-` should be safe, even in a future where we accept RESP3 formatted commands it's not likely that clients will be sending errors to servers.

Reply:
If we use RESP3 in AOF, maybe timestamp looks like that. But it brings much complexity for `processMultibulkBuffer`. i am sure we will support in the future.
```
*3
|1
+ts
:1632711499
$3
set
$1
a
$1
b
```
The reason why I use `#` is that  `#` is some languages' annotation. Currently, RESP3 already used many normal characters. i have no idea which character to use.

Reply:
Maybe use / instead? Kind of like a comment?

Reply:
`/` is a comment? you're referring to `//`?
I feel this discussion is a dead end.. let's stick to `#`